pybop/parameters/parameter.py

```diff
@@ -381,6 +405,43 @@ def get_bounds_for_plotly(self):
         return bounds
 
     def as_dict(self, values=None) -> Dict:
+        """
+        Parameters
+        ----------
+        values : list or str, optional
+            A list of parameter values or one of the strings "initial" or "true" which can be used
+            to obtain a dictionary of parameters.
+
+        Returns
+        -------
+        Inputs
+            A parameters dictionary.
+        """
         if values is None:
             values = self.current_value()
+        elif isinstance(values, str):
+            if values == "initial":
+                values = self.initial_value()
+            elif values == "true":
+                values = self.true_value()
```

Comment:
I'm in favour of the previous API. Having `parameters.current()`, `parameters.initial()`. and `parameters.true()` would be my ideal preference though. Having these as methods is very helpful for tab completions in the REPL or IDE of choice and avoids potential bugs in the string argument (i.e "initial", "Initial", "intial", etc.)

Reply:
I think it is important to signpost that the output is a dictionary, as opposed to the output of `initial_value`. Perhaps `initial_dict` etc.? 

Should probably also update the return type to `Inputs` rather than `Dict`.